### PR TITLE
[cxx] Remove two excess toplevel consts in structs, one a C++ problem, one very obscure/hypothetical.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/metadata/object-internals.h [hunk 1]
```diff
@@ -356,7 +356,7 @@ struct _MonoIUnknownVTable
 
 struct _MonoIUnknown
 {
-	MonoIUnknownVTable const * const vtable;
+	const MonoIUnknownVTable *vtable;
```
Comment (jaykrell) [inline mono/metadata/object-internals.h]: Hypothetical problem.

File: mono/utils/mono-merp.c [hunk 2]
```diff
@@ -97,7 +97,7 @@ typedef struct {
 	const char *osVersion; 
 	int uiLidArg; // Application LCID 
 
-	const char systemModel [100];
+	char systemModel [100];
```
Comment (jaykrell) [inline mono/utils/mono-merp.c]: Real problem.

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 Coop Suspend

**Comment by jaykrell** [comment]
@monojenkins build Linux x64

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

**Comment by jaykrell** [comment]
Review?

File: mono/utils/mono-merp.c (hunk 2 above)
Comment (lambdageek) [reply mono/utils/mono-merp.c]: What's the problem here? all uses of systemModel are either okay with `const` or cast constness away.

I'm not opposed to this change, but I'm curious what it's fixing.

Comment (jaykrell) [reply mono/utils/mono-merp.c]: Error message is in the commit. Having const members requires initializing them, once — or multiple times and casting away const as you say. There may be another fix.

Comment (jaykrell) [reply mono/utils/mono-merp.c]: I’m really torn on this. More const seems good, even if interface-only. Lack of these errors in general suggests it is rare. And not idiomatic. But still good. Rare != bad, often the opposite. New/better implies rare.

The secondary reason is a recent msvc compiler regression and will not stand long, did ship, does not hit us, probably never will.

**Pull request merged**